### PR TITLE
Adds assert messages to std.algorithm.sorting

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -617,7 +617,8 @@ size_t pivotPartition(alias less = "a < b", Range)
 (Range r, size_t pivot)
 if (isRandomAccessRange!Range && hasLength!Range && hasSlicing!Range && hasAssignableElements!Range)
 {
-    assert(pivot < r.length || r.length == 0 && pivot == 0);
+    assert(pivot < r.length || r.length == 0 && pivot == 0, "pivot must be"
+        ~ " less then the length of r or r must be empty and pivot zero");
```
Comment (thewilsonator) [inline std/algorithm/sorting.d]: > less then

less th_a_n

File: std/algorithm/sorting.d [hunk 2]
```diff
@@ -1682,7 +1683,7 @@ private void shortSort(alias less, Range)(Range r)
             break;
     }
 
-    assert(r.length >= 6);
+    assert(r.length >= 6, "r must be longer than 5");
```
Comment (thewilsonator) [inline std/algorithm/sorting.d]: r must have more than 5 elements

File: std/algorithm/sorting.d [hunk 3]
```diff
@@ -1740,7 +1741,7 @@ Sorts the first 5 elements exactly of range r.
 */
 private void sort5(alias lt, Range)(Range r)
 {
-    assert(r.length >= 5);
+    assert(r.length >= 5, "r must be longer than 4 elements");
```
Comment (thewilsonator) [inline std/algorithm/sorting.d]: ditto

File: std/algorithm/sorting.d (hunk 1 above)
Comment (burner) [reply std/algorithm/sorting.d]: this always happens to me.

Comment (burner) [reply std/algorithm/sorting.d]: fixed

File: std/algorithm/sorting.d (hunk 2 above)
Comment (burner) [reply std/algorithm/sorting.d]: fixed

File: std/algorithm/sorting.d (hunk 3 above)
Comment (burner) [reply std/algorithm/sorting.d]: fixed

File: std/algorithm/sorting.d (hunk 1 above)
Comment (thewilsonator) [reply std/algorithm/sorting.d]: No it isn't.

File: std/algorithm/sorting.d (hunk 2 above)
Comment (thewilsonator) [reply std/algorithm/sorting.d]: ditto

File: std/algorithm/sorting.d (hunk 1 above)
Comment (burner) [reply std/algorithm/sorting.d]: it helps to commit, my bad

**Comment by burner** [comment]
thanks, sorry for the noise

**Comment by WalterBright** [comment]
@burner @thewilsonator please do not use `format` in assert messages. `format` expands to a truly epic blizzard of template instantiations, which causes lots of problems with debug and unittest builds. It also causes problems by making problems hard to isolated, due to depending on most of the rest of Phobos.

`assert` messages should be as lightweight as possible, and should not instantiate templates. `sprintf` can be used for formatting.

**Pull request merged**